packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -519,19 +535,41 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
   }
 
   void _initMouseGestureRecognizer() {
-    _gestureRecognizers[TapAndPanGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapAndPanGestureRecognizer>(
-          () => TapAndPanGestureRecognizer(debugOwner:this, supportedDevices: <PointerDeviceKind>{ PointerDeviceKind.mouse }),
-          (TapAndPanGestureRecognizer instance) {
-        instance
-          ..onTapDown = _startNewMouseSelectionGesture
-          ..onTapUp = _handleMouseTapUp
-          ..onDragStart = _handleMouseDragStart
-          ..onDragUpdate = _handleMouseDragUpdate
-          ..onDragEnd = _handleMouseDragEnd
-          ..onCancel = _clearSelection
-          ..dragStartBehavior = DragStartBehavior.down;
-      },
-    );
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.iOS:
+        _gestureRecognizers[TapAndHorizontalDragGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapAndHorizontalDragGestureRecognizer>(
```

Comment:
Can you add some comment on why we use tap and horizontal drag here? also is fuchsia meant to be here?

Reply:
Within `TextSelectionGestureDetector` we default to grouping fuchsia with Android if the behavior is unknown.